release/release.go

```diff
@@ -121,11 +121,107 @@ func buildLinuxPackages() {
+
+
+	// set up build directory.
+	log.Printf("create rpm directory tree\n")
+	rpmBuildDir := "rpm_build"
+	check(os.RemoveAll(rpmBuildDir), "removeAll "+rpmBuildDir)
+	check(os.MkdirAll(rpmBuildDir, os.ModePerm), "mkdirAll "+rpmBuildDir)
+	check(os.Chdir(rpmBuildDir), "cd to "+rpmBuildDir)
+	oldCwd, err := os.Getwd()
+	check(err, "get current directory")
+	defer os.Chdir(oldCwd)
+	// we'll want to go back to the original directory, just in case.
+	// build the release dir.
+	// The goal here is to set up  directory with the following structure:
+	// rpmbuild/
+	// |----- SOURCES/
+	// |         |----- mongodb-database-tools.tar.gz:
+	//                       |
+	//                      mongodb-database-tools/
+	//                               |------ usr/
+	//                               |-- bin/
+	//                               |    |--- bsondump
+	//                               |    |--- mongo*
+	//                               |-- share/
+	//                                      |---- doc/
+	//                                             |----- mongodb-database-tools/
+	//                                                              |--- staticFiles
+
+	home := os.Getenv("HOME")
+	// create tar file
+	log.Printf("tarring necessary files")
+	createTar := func() {
+		staticFilesPath := ".."
+		binariesPath := filepath.Join("..", "bin")
+		sources := filepath.Join(home, "rpmbuild", "SOURCES")
+		check(os.MkdirAll(sources, os.ModePerm), "create " + sources)
+		archiveFile, err := os.Create(filepath.Join(sources, mdt + ".tar.gz"))
+		check(err, "create archive file")
+		defer archiveFile.Close()
+
+		gw := gzip.NewWriter(archiveFile)
+		defer gw.Close()
+
+		tw := tar.NewWriter(gw)
+		defer tw.Close()
+
+		for _, name := range staticFiles {
+			log.Printf("adding %s to tarball\n", name)
+			src := filepath.Join(staticFilesPath, name)
+			dst := filepath.Join(mdt, "usr", "share", "doc", mdt, name)
+			addToTarball(tw, dst, src)
+		}
+
+		for _, name := range binaries {
+			log.Printf("adding %s to tarball\n", name)
+			src := filepath.Join(binariesPath, name)
+			dst := filepath.Join(mdt, "usr", "bin", name)
+			addToTarball(tw, dst, src)
+		}
+	}
+	createTar()
+
+	p, err := platform.Get()
+	check(err, "get platform")
+	specFile := mdt + ".spec"
+
+	rpmVersion := getRPMVersion(getVersion())
+	createSpecFile := func() {
+		log.Printf("create spec file")
+		f, err := os.Create(specFile)
+		check(err, "create spec")
+		defer f.Close()
+
+		// get the control file content.
+		contentBytes, err := ioutil.ReadFile(filepath.Join("..", "installer", "rpm", specFile))
+		content := string(contentBytes)
+		check(err, "reading spec file content")
+		content = strings.Replace(content, "@TOOLS_VERSION@", rpmVersion, -1)
+		content = strings.Replace(content, "@ARCHITECTURE@", p.Arch, 1)
+		_, err = f.WriteString(content)
+		check(err, "write content to spec file")
+	}
+	createSpecFile()
+
+	outputFile := mdt + "-" + rpmVersion + "-2." + p.Arch + ".rpm"
```

Comment:
why the `-2.`? Just rpm convention?

Reply:
The 2 is an rpm convention, and I don't know why. It's worth looking up, I think. But info on rpms is a lot harder to find than .deb files.

Reply:
The 2 is actually the release. It's interesting because we were doing the same thing (just hardcoding 2) when I did this in the past.